p2p/transport/webrtc/stream_read.go

```diff
@@ -0,0 +1,166 @@
+package libp2pwebrtc
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+	"time"
+
+	pb "github.com/libp2p/go-libp2p/p2p/transport/webrtc/pb"
+	"github.com/libp2p/go-msgio/pbio"
+	"github.com/pion/datachannel"
+)
+
+type (
+	webRTCStreamReader struct {
+		stream *webRTCStream
+		state  *webRTCStreamReaderState
+
+		deadline    time.Time
+		deadlineMux sync.Mutex
+
+		closeOnce sync.Once
+	}
+
+	webRTCStreamReaderState struct {
+		sync.Mutex
```

Comment:
Why is this mutex needed. Concurrent calls to `Read` are not permitted anyway.

Reply:
It's my first dev contribution to libp2p, so not sure about the implicit contracts that transports agree to. If that's the case we can indeed simplify all that a lot. Does the same go for write? 

That said the language does not give guarantees here though, but if you have this as a contract that all agree to, it's something I suppose.

Reply:
Removed the mutex usage here!